pymeasure/instruments/keysight/keysightDSOX1102G.py

```diff
@@ -0,0 +1,559 @@
+        self.write(":stop")
+
+    def single(self):
+        """ Causes the instrument to acquire a single trigger of data.
+        This is the same as pressing the Single key on the front panel. """
+        self.write(":single")
+
+    _digitize = Instrument.setting(
+        ":DIGitize %s",
+        """ Acquire waveforms according to the settings of the :ACQuire commands and specified source,
+         as a string parameter that can take the following values: "channel1", "channel2", "function",
+         "math", "fft", "abus", or "ext". """,
+        validator=strict_discrete_set,
+        values={"channel1": "CHAN1", "channel2": "CHAN2", "function": "FUNC", "math": "MATH",
+                "fft": "FFT", "abus": "ABUS", "ext": "EXT"},
+        map_values=True
+    )
+
+    def digitize(self, source: str):
+        """ Acquire waveforms according to the settings of the :ACQuire commands. Ensure a delay
+        between the digitize operation and further commands, as timeout may be reached before
+        digitize has completed.
+        :param source: "channel1", "channel2", "function", "math", "fft", "abus", or "ext"."""
+        self._digitize = source
+
+    waveform_points_mode = Instrument.control(
+        ":waveform:points:mode?", ":waveform:points:mode %s",
+        """ A string parameter that sets the data record to be transferred with the waveform_data
+         method. Can be "normal", "maximum", or "raw".""",
+        validator=strict_discrete_set,
+        values={"normal": "NORM", "maximum": "MAX", "raw": "RAW"},
+        map_values=True
+    )
+    waveform_points = Instrument.control(
+        ":waveform:points?", ":waveform:points %d",
+        """ An integer parameter that sets the number of waveform points to be transferred with
+        the waveform_data method. Can be any of the following values: 
+        100, 250, 500, 1000, 2 000, 5 000, 10 000, 20 000, 50 000, 62 500.
+        
+        Note that the oscilloscope may provide less than the specified nb of points. """,
+        validator=strict_discrete_set,
+        values=[100, 250, 500, 1000, 2000, 5000, 10000, 20000, 50000, 62500]
+    )
+    waveform_source = Instrument.control(
+        ":waveform:source?", ":waveform:source %s",
+        """ A string parameter that selects the analog channel, function, or reference waveform 
+        to be used as the source for the waveform methods. Can be "channel1", "channel2", "function", 
+        "fft", "wmemory1", "wmemory2", or "ext".""",
+        validator=strict_discrete_set,
+        values={"channel1": "CHAN1", "channel2": "CHAN2", "function": "FUNC", "fft": "FFT",
+                "wmemory1": "WMEM1", "wmemory2": "WMEM2", "ext": "EXT"},
+        map_values=True
+    )
+    waveform_format = Instrument.control(
+        ":waveform:format?", ":waveform:format %s",
+        """ A string parameter that controls how the data is formatted when sent from the 
+        oscilloscope. Can be "ascii".""",
```

Comment:
'WORD' and 'BYTE' formats are also available. I do believe I had not included them here because of the issue with querying binary values (as per your other comments).

I'll try to work with instrument.binary_values and might be able to allow 'WORD' and 'BYTE' as waveform types here.